hubgrep/lib/hosting_service_interfaces/_hosting_service_interface.py

```diff
@@ -135,19 +137,43 @@ def normalize_url(url) -> str:
 
 
 class HostingServiceInterfaceResponse:
-    """ Wrapper class containing data and references from a hosting-service response. """
+    """
+    Wrapper class containing a hosting_interface,
+    the interfaces response and potential search results.
+
+    Used in the search jinja templates.
+    """
+
     hosting_service_interface: HostingServiceInterface
     response: CachedResponse
     search_results: List[SearchResult]
 
-    def __init__(self,
-                 hosting_service_interface: HostingServiceInterface,
-                 response: CachedResponse,
-                 search_results: List[SearchResult]):
+    def __init__(
+        self,
+        hosting_service_interface: HostingServiceInterface,
+        response: CachedResponse,
+        search_results: List[SearchResult],
+    ):
         self.hosting_service_interface = hosting_service_interface
         self.response = response
         self.search_results = search_results
 
     @property
     def succeeded(self) -> bool:
         return self.response.success
+
+    def get_admin_error_message(self):
+        return self.response.error_msg
+
+    def get_user_error_message(self):
+        """
+        get an error message to display to a user.
+        """
+        if self.response.status_code == -1:
+            return "Request took to long (or other connection error)"
+        elif 400 <= self.response.status_code <= 499:
+            return "Client error"
+        elif 500 <= self.response.status_code <= 599:
+            return "Server error"
+        else:
+            return "Unknown error"
```

Comment:
My stance is we shouldn't output "Unknown error" or specifics of what happens in the backend for parties that are not involved with underlaying things such as crawlers, hosting-services, requests or whatever.

I think non-auth'ed users should only see messages that are actively helpful for them that they can take action on, on the page (for example, "things dont work right now, come back later" is an "action" in that sense".

For instance "Request" is a word that might not mean anything to an actual user (even a dev) because they have no idea what/which request, even if they know what it means. Same goes for Server/Client - these aren't things a not logged in user need to see, or has any use for.

The phrase "Took too long." in some form MIGHT be useful, as it may prompt someone to try it again and hit search - but adding "or something else" takes away from being a helpful directive.

For our "-1" status its only timeout right? and if those requests fail with a statuscode we will also use that code instead. I don't think "connection error" applies, or?

So basically:
"not logged in = no underlaying information that isn't directly helpful to the user RIGHT NOW".
So when you're un-auth'ed you only get "something went wrong", and at most "something went wrong - but try again".

Reply:
We havent gotten into this yet, but we should clearly mark our own statuses like -1, so that its specific internally.

Reply:
Also =D

If we move the error msg creation into the code, that should be the only way it happens (and I think it should happen outside the template like you started here) - just, theres still stuff in the template for it. I would want to move that out too then.

Reply:
> My stance is we shouldn't output "Unknown error" or specifics of what happens in the backend for parties that are not involved with underlaying things such as crawlers, hosting-services, requests or whatever.

yea, but here its actually an unknown error, since http codes are only defined to 5xx - everything above shouldnt exist, so its really just there to have a non-empty "else" branch.
(-> https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)

Reply:
> Also =D
> 
> If we move the error msg creation into the code, that should be the only way it happens (and I think it should happen outside the template like you started here) - just, theres still stuff in the template for it. I would want to move that out too then.

i think the decision which error message to show shoulnt live in this class, since its specific to the request. maybe we do that in the route? (but i dont find it too ugly in the template either.. seems fine to me the way it is)?

Reply:
> We havent gotten into this yet, but we should clearly mark our own statuses like -1, so that its specific internally.

the code isnt shown to users, right? only admins?
(also, i think it kind of makes sense - maybe just a comment in the code and in the docs somewhere for that? "-1 means we dont have a status code, since we didnt get an answer")